Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/service/decorator/HttpHeaderServiceEndpointDecorator.java

```diff
@@ -0,0 +1,144 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.messaging.service.decorator;
+
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.CONNECTCustomHttpHeadersType;
+import gov.hhs.fha.nhinc.messaging.service.ServiceEndpoint;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.nhinclib.NullChecker;
+import gov.hhs.fha.nhinc.properties.PropertyAccessException;
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import java.util.HashMap;
+import java.util.Set;
+import javax.xml.ws.BindingProvider;
+import org.apache.cxf.transports.http.configuration.ConnectionType;
+import org.apache.cxf.transports.http.configuration.HTTPClientPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author jassmit
+ * @param <T>
+ */
+public class HttpHeaderServiceEndpointDecorator<T> extends ServiceEndpointDecorator<T> {
+
+    private final AssertionType assertion;
+
+    private static final Logger LOG = LoggerFactory.getLogger(HttpHeaderServiceEndpointDecorator.class);
+
+    public HttpHeaderServiceEndpointDecorator(ServiceEndpoint<T> decoratedEndpoint, final AssertionType assertion) {
+        super(decoratedEndpoint);
+        this.assertion = assertion;
+    }
+
+    @Override
+    public void configure() {
+        super.configure();
+        final HTTPClientPolicy httpClientPolicy = getHTTPClientPolicy();
+        final BindingProvider bindingProvider = (BindingProvider) getPort();
+
+        //Clear http connection value.
+        httpClientPolicy.setConnection(null);
+        if (hasKeepAliveForConnection()) {
+            httpClientPolicy.setConnection(ConnectionType.KEEP_ALIVE);
+        }
+
+        bindingProvider.getRequestContext().remove(NhincConstants.CUSTOM_HTTP_HEADERS);
+        HashMap<String, String> httpHeaders = getCustomHeaders();
+
+        if (httpHeaders != null && !httpHeaders.isEmpty()) {
+            bindingProvider.getRequestContext().put(NhincConstants.CUSTOM_HTTP_HEADERS, httpHeaders);
+        }
+    }
+
+    private boolean hasKeepAliveForConnection() {
+        String keepAlive = assertion.getKeepAlive();
+        if (NullChecker.isNullish(keepAlive)) {
+            keepAlive = getKeepAliveProperty();
+        }
+
+        return NullChecker.isNotNullish(keepAlive)
```

Comment:
Yes, the property could possibly not be set if there is a property access exception.